providers/dns/liquidweb/liquidweb.go

```diff
@@ -0,0 +1,141 @@
+package liquidweb
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/go-acme/lego/v3/challenge/dns01"
+	"github.com/go-acme/lego/v3/platform/config/env"
+	lw "github.com/liquidweb/liquidweb-go/client"
+	"github.com/liquidweb/liquidweb-go/network"
+)
+
+// Config is used to configure the creation of the DNSProvider
+type Config struct {
+	URL                string
+	Username           string
+	Password           string
+	Zone               string
+	Timeout            time.Duration
+	PollingInterval    time.Duration
+	PropagationTimeout time.Duration
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider
+func NewDefaultConfig() *Config {
+	config := &Config{
+		Timeout:         env.GetOrDefaultSecond("LIQUID_WEB_TIMEOUT", 1*time.Minute),
+		PollingInterval: env.GetOrDefaultSecond("LIQUID_WEB_POLLING_INTERVAL", 2*time.Second),
+		PropagationTimeout: env.GetOrDefaultSecond("LIQUID_WEB_PROPAGATION_TIMEOUT",
+
+			2*time.Minute),
+	}
+
+	return config
+}
+
+// DNSProvider is an implementation of the acme.ChallengeProvider interface
+// that uses Liquid Web's REST API to manage TXT records for a domain.
+type DNSProvider struct {
+	config      *Config
+	recordIDs   map[string]int
+	recordIDsMu sync.Mutex
+	client      *lw.API
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for Liquid Web.
+func NewDNSProvider() (*DNSProvider, error) {
+	config := NewDefaultConfig()
+	config.URL = env.GetOrDefaultString("LIQUID_WEB_URL", "")
+	config.Username = env.GetOrDefaultString("LIQUID_WEB_USERNAME", "")
+	config.Password = env.GetOrDefaultString("LIQUID_WEB_PASSWORD", "")
+	config.Zone = env.GetOrDefaultString("LIQUID_WEB_ZONE", "")
```

Comment:
Could you follow this pattern:
https://github.com/go-acme/lego/blob/e820634e0e43baff0ef61728cb8379f804913b72/providers/dns/auroradns/auroradns.go#L47-L59